### DISCUSSION TITLE
Extension .yml VS .yaml

### DISCUSSION THREAD

**Discussion by bessarabov** [umputun/spot#79] [General]
The default playbook is `spot.yml` with 3 letter `.yml` extension.

And there are several files in the repo that also have `.yml` extension.

But it looks like the official extension for YAML is 4 letter: `.yaml`. This is stated on the official site https://yaml.org/faq.html :

```
Is there an official extension for YAML files?
Please use ".yaml" when possible.
```

What do you think about changing everywhere from `.yml` to `.yaml` ?

**Comment by umputun** [comment]
I don't think spot will have any trouble treating .yaml as yaml file. I mean, unless it is .toml, it will be parsed as yml, regardless of the extension. 

Changing the default extension in the code doesn't feel right to me. I see .yml everywhere and not .yaml

**Comment by bessarabov** [comment]
Ha. This is pretty funny. I felt that .yaml is more commonly used, than .yml, but it looks like you are right — I've checked several projects, most of them work fine with both .yaml & .yml files (not all), but a lot of them use .yml fist. and .yaml as second choice:

.yml team:
 * github actions — https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
 * gitlab ci — https://docs.gitlab.com/ee/ci/yaml/
 * circle ci — https://circleci.com/docs/introduction-to-yaml-configurations/
 * travic ci — https://docs.travis-ci.com/user/customizing-the-build/
 * jekyl — https://jekyllrb.com/docs/datafiles/

.yaml team:
 * ansible — https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yaml_inventory.html
 * kubernetes — https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/
 * docker-compose — https://docs.docker.com/compose/compose-file/03-compose-file/
 * hugo — https://gohugo.io/getting-started/configuration/
 * home assistant — https://www.home-assistant.io/docs/configuration/
 * puppet — https://www.puppet.com/docs/bolt/latest/writing_yaml_plans.html
 
 I think that I'm going to close this discussion. Thank you.

**Reply by umputun** [reply]
nah, docker-compose my team as well ;) this is from the official docs

![image](https://github.com/umputun/spot/assets/535880/ab2796ec-1c51-4a18-ab26-294410a5f127)

**Reply by bessarabov** [reply]
🤣 

![Screen Shot 2023-05-09 at 22 36 18](https://github.com/umputun/spot/assets/47263/8e905741-02c9-4a39-90aa-29508cb532b5)
